src/SelfDestructLinkButton/SelfDestructLinkButton.js

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+
+import React, { useState, useRef } from 'react';
+import PropTypes from 'prop-types';
+import {
+  Button,
+  CircularProgress
+} from '@material-ui/core';
+import useStyles from './style';
+
+const STATUS = {
+  IDLE: 'IDLE',
+  GENERATING: 'GENERATING',
+  READY: 'READY'
+};
+
+const defaultLabels = {
+  generateLink: 'Generate link',
+  download: 'Download'
+};
+
+const SelfDestructLinkButton = (props) => {
+  const classes = useStyles();
+  const {
+    generate,
+    download,
+    labels,
+    timeout
+  } = props;
+
+  const [status, setStatus] = useState(STATUS.IDLE);
+  const [downloadLink, setDownloadLink] = useState(null);
+  const timeoutRef = useRef(null);
+
+  const reset = () => {
+    setStatus(STATUS.IDLE);
+    setDownloadLink(null);
+    if (timeoutRef.current) {
+      clearTimeout(timeoutRef.current);
```

Comment:
What's the purpose of clearTimeout?

Reply:
Clearing the timeout prevents a reset to be triggered if the user clicks several time on the button. For instance, if the link-destruction timeout is set to 20 seconds, and the user follows these steps:
- generate a link
- click on the link but cancel the download
- generate a new link
- the new link may be deleted by the 1st timeout if it hasn't been cleared